### PR TITLE
doc: scripts: Updates for pyocd v0.11.1

### PR DESCRIPTION
**Pull request by MaureenHelm**
pyocd recently added support for python 3 so we can now remove the
python 2 package requirements. It also merged Zephyr thread awareness
upstream, so we can remove the reference to my pyocd pull request.

Tested debugging and flashing on Linux and Windows.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

### CODE REVIEW

File: scripts/requirements.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@ ply==3.10
 hub==2.0
 gitlint
 pyelftools==0.24
+pyocd
```
Comment (dbkinder) [inline scripts/requirements.txt]: should this be:

    pyocd>=0.11.0

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, question for requirements.txt change

File: doc/application/application.rst [hunk 2]
```diff
@@ -674,9 +674,8 @@ Create a Debugger Configuration
 RTOS Awareness
 ==============
 
-Experimental support for Zephyr RTOS awareness is implemented in `pyOCD PR
-#333`_. It is compatible with GDB PyOCD Debugging in Eclipse, but you must
-download this pull request and build pyOCD from source. You must also enable
+Support for Zephyr RTOS awareness is implemented in `pyOCD v0.11.0`_ and later.
```
Comment (dbkinder) [inline doc/application/application.rst]: PR title says v0.11.1 so should this say 0.11.1 as well?

**Review by nashif — APPROVED** [review]

File: scripts/requirements.txt (hunk 1 above)
Comment (MaureenHelm) [reply scripts/requirements.txt]: I don't think so. Not including the version specifier means install the latest version, which as of today is 0.11.1.

File: doc/application/application.rst (hunk 2 above)
Comment (MaureenHelm) [reply doc/application/application.rst]: This feature was added in v0.11.0, but I tested with v0.11.1.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**